packages/datetime/src/common/dateUtils.ts

```diff
@@ -98,6 +98,12 @@ export function getDateBetween(dateRange: DateRange) {
 }
 
 export function getDateTime(date: Date, time: Date) {
-    return new Date(date.getFullYear(), date.getMonth(), date.getDate(),
+    if (date === null) {
+        return null;
```

Comment:
This is used for the `DatePicker` within `DateTimePicker` and `null` / `undefined` will differentiate between controlled / uncontrolled.